onset_maproom/agronomy.py

```diff
@@ -0,0 +1,535 @@
+        ks = single_stress_coeff(
+            sminit0,
+            taw,
+            raw.isel({time_coord: 0}, missing_dims="ignore"),
+            time_coord=time_coord,
+        ).squeeze(time_coord)
+    # Create or Initialize Kc
+    if kc_params is None:
+        kc = xr.ones_like(water_balance.et)
+    else:
+        if p_d is not None:
+            kc = kc_interpolation(p_d, kc_params, time_coord=time_coord)
+        else:
+            p_d_find = planting_date(
+                sminit0, sm_threshold, time_coord=time_coord
+            ).expand_dims(dim=time_coord)
+            kc = kc_interpolation(p_d_find, kc_params, time_coord=time_coord)
+    # Initializaing sm, drainage, et_crop, et_crop_red
+    if time_coord in kc.dims:
+        kc0 = kc.sel({time_coord: water_balance.soil_moisture[time_coord][0]})
+    else:
+        kc0 = kc
+    water_balance.et_crop[{time_coord: 0}] = crop_evapotranspiration(
+        water_balance.et.isel({time_coord: 0}).expand_dims(dim=time_coord),
+        kc0,
+        time_coord=time_coord,
+    ).squeeze(time_coord)
+    water_balance.et_crop_red[{time_coord: 0}] = reduce_crop_evapotranspiration(
+        water_balance.et_crop
+        .isel({time_coord: 0}).expand_dims(dim=time_coord),
+        ks,
+    ).squeeze(time_coord)
+    water_balance.soil_moisture[{time_coord: 0}] = (
+        sminit
+        + water_balance.peffective.isel({time_coord: 0})
+        - water_balance.et_crop_red.isel({time_coord: 0})
+    ).clip(min=0)
+    water_balance.drainage[{time_coord: 0}] = (
+        water_balance.soil_moisture.isel({time_coord: 0})
+        - taw
+    ).clip(min=0)
+    water_balance.soil_moisture[{time_coord: 0}] = (
+        water_balance.soil_moisture.isel({time_coord: 0})
+        - water_balance.drainage.isel({time_coord: 0})
+    )
+    # Looping on time_coord
+    # Get time_coord info
+    time_coord_size = water_balance.peffective[time_coord].size
+    for i in range(1, time_coord_size):
+        if rho is not None:
+            ks = single_stress_coeff(
+                water_balance.soil_moisture
+                .isel({time_coord: i - 1}).expand_dims(dim=time_coord),
+                taw,
+                raw,
+                time_coord=time_coord,
+            ).squeeze(time_coord)
+        if kc_params is not None and p_d is None:
+            p_d_iter = planting_date(
+                water_balance.soil_moisture
+                .isel({time_coord: i - 1}).expand_dims(dim=time_coord),
+                sm_threshold,
+                time_coord=time_coord,
+            )
+            # it's desired and useful to update Kc if
+            # p_d was NaT and is now found
+            kc_updates = np.isnat(p_d_find) and not np.isnat(p_d_iter)
+            p_d_find = p_d_find.where(
+                lambda x: not np.isnat(x),
+                other = p_d_iter + p_d_iter[time_coord] - p_d_find[time_coord]
+            )
+            kc = kc.where(
+                not kc_updates,
```

Comment:
Should be `~kc_updates`. Again, the fact that this went unnoticed probably means that the tests are insufficient.